pulsar-common/src/main/java/org/apache/pulsar/common/naming/TopicName.java

```diff
@@ -79,11 +78,7 @@ public static TopicName get(String domain, String tenant, String cluster, String
     }
 
     public static TopicName get(String topic) {
-        try {
-            return cache.get(topic);
-        } catch (ExecutionException | UncheckedExecutionException e) {
-            throw (RuntimeException) e.getCause();
-        }
+        return cache.computeIfAbsent(topic, k -> new TopicName(k));
```

Comment:
computeIfAbsent is most often pessimistic in its locking, whereas Caffeine does a lock-free read first before falling back to a computation. Since caffeine is built on top of CHM and your readers will often block each other, something seems very amiss with your benchmark results being so vastly different in performance.

Reply:
Answered [here](https://github.com/apache/pulsar/pull/23052#issuecomment-2241487669)